### ISSUE TITLE
pilot process taking 100% cpu time

### ISSUE THREAD

**Issue by dsschult** [WIPACrepo/iceprod#119]
Looks like the condition variable is the culprit.  Waking up far too often.

Here is one second of activity captured using `strace`:

```
11:48:51.020874 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.020972 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.071331 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.071387 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.071421 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.071488 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.121811 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.121863 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.121896 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.121943 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.121977 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.172827 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.223276 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.223406 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.223511 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.223675 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.223797 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.274183 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.274229 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.274284 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.324635 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.324682 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.324713 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.324795 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.375162 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.375215 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.425608 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.425660 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.425729 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.476058 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.476138 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.476202 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.476287 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.476342 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.526719 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.526796 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.526841 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.577063 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.577159 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.577243 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.577367 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.577466 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.577540 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.627927 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.628012 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.628066 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.628131 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.628197 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.628239 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.678592 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.678645 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.678679 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.678744 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.678780 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.729043 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.729132 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.729209 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.729273 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.729348 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.729407 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.783019 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.783094 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.783146 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.833496 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.833577 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.883902 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.883974 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.884046 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.884109 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.884163 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.884267 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.884311 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.934648 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.934707 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.934748 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.934795 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.934829 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
11:48:51.985227 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = -1 EAGAIN (Resource temporarily unavailable)
11:48:51.985359 futex(0x16f9df0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
11:48:51.985508 futex(0x16f9df0, FUTEX_WAKE_PRIVATE, 1) = 1
11:48:51.985611 select(0, NULL, NULL, NULL, {0, 50000}) = 0 (Timeout)
```

**Comment by dsschult** [comment]
Looks like that only changed the problem.  `tornado` isn't efficient at handling condition timeouts either, for some reason: 

```
16:06:55.888987 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889061 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889139 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889213 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889297 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889384 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889457 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889589 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889663 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889742 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889813 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889884 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.889956 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890033 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890112 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890184 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890255 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890387 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890470 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890615 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890689 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890764 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890835 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890911 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.890985 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891053 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891122 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891193 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891271 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891351 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891422 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891502 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891629 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891704 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891773 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891848 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.891924 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892007 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892078 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892148 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892218 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892293 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892368 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892440 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892566 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892644 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892724 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892794 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892864 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.892933 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.893048 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.893128 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.893197 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.893269 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.893343 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
16:06:55.893418 wait4(11886, 0x7ffd0d73a244, WNOHANG, NULL) = 0
```

**Comment by dsschult** [comment]
I think this is fixed in 8028ad387d234d2e7d0a8bc3c24d60b18a6856ac
